astropy/coordinates/representation.py

```diff
@@ -223,6 +223,24 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
```

Comment:
This is needed because of some checks the coordinate machinery does initialising frames - two frames with identical ```obsgeoloc``` attributes need to satisfy ```np.any(obsgeoloc1 == obsgeoloc2)```

However, that would also  be alright if this returned an array...